monkey/monkey_island/cc/ui/src/components/ui-components/HideInput.js

```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import {InputGroup, FormControl} from 'react-bootstrap';
+import '../../styles/components/HideInput.scss'
+
+class HideInput extends React.PureComponent {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      hidden: false
+    };
+    this.toggleShow = this.toggleShow.bind(this);
+  }
+
+  toggleShow() {
+    this.setState({hidden: ! this.state.hidden});
+  }
+
+  onChange(e) {
```

Comment:
Again, my javascript is rusty. Does this also need a call to `bind()` or equivalent?

Reply:
https://newbedev.com/why-and-when-do-we-need-to-bind-functions-and-eventhandlers-in-react
Apparently there is couple ways of binding, in which I have use the third one. 

`onChange={(event) => this.onChange(event)}`